### PR TITLE
[DependencyInjection] Disable dumping the container as XML in debug mode

### LINKED ISSUE #17952: [FrameworkBundle] Allow disabling dumping of container to XML to improv…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#49487
| PR author(s) | @ruudk
| Merged in    | 6.3

### PR DESCRIPTION
**Pull request by ruudk**
Closes #17952

### CODE REVIEW

File: performance.rst [hunk 1]
```diff
@@ -217,6 +217,44 @@ deployment process too):
   used in your application and prevents Composer from scanning the file system for
   classes that are not found in the class map. (see: `Composer's autoloader optimization`_).
 
+Disable dumping the container as XML in debug mode
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (OskarStark) [inline performance.rst]:
```suggestion
Disable Dumping the Container as XML in Debug Mode
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by OskarStark — APPROVED** [review]
After my comment

File: performance.rst [hunk 2]
```diff
@@ -217,6 +217,44 @@ deployment process too):
   used in your application and prevents Composer from scanning the file system for
   classes that are not found in the class map. (see: `Composer's autoloader optimization`_).
 
+Disable Dumping the Container as XML in Debug Mode
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In debug mode, Symfony compiles the :doc:`service container </service_container>`
+into an XML file. This XML file is used by various commands such as
+``debug:container`` and ``debug:autowiring``. You can disable dumping of the XML
```
Comment (94noni) [inline performance.rst]: Perhaps write why one would want to disable it
Time saving at cache clear etc

Comment (ruudk) [reply performance.rst]: Made some changes, do you like this more?

Comment (94noni) [reply performance.rst]: yes it is worth documenting the why, not only the how :)
so this new sentence is better imho 👍🏻

**Review by 94noni — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thanks Ruud!

**Pull request merged**